geps/gep-3155/index.md

```diff
@@ -0,0 +1,266 @@
+  //
+  // Support: Core
+  DefaultClientCertificateInheritanceMode ClientCertificateInheritanceMode = "Default"
+
+  // Override determines the certificate specified in ClientCertificateRef field will be used.
+  //
+  // Support: Implementation-specific
+  OverrideClientCertificateInheritanceMode ClientCertificateInheritanceMode = "Override"
+
+  // Disable determines the certificate will be not provided to the backend, even if it is
+  // configured at the gateway level and requested by the backend.
+  //
+  // Support: Implementation-specific
+  DisableClientCertificateInheritanceMode ClientCertificateInheritanceMode = "Disable"
+)
+
+type BackendTLSPolicySpec {
+  // ClientCertificateMode determines the inheritance mode for client certifacate.
+  //
+  // If ClientCertificateMode is unset, the value assumed is Default.
+  ClientCertificateMode ClientCertificateInheritanceMode `json:"clientCertificateMode,omitempty"`
+
+  // ClientCertificateRef is a reference to an object that contains a Client
+  // Certificate.
+  //
+  // This value takes effect and is required only if the ClientCertificateMode is set to 
+  // Override, otherwise it is ignored.
+  //
+  // References to a resource in different namespace are invalid UNLESS there
+  // is a ReferenceGrant in the target namespace that allows the certificate
+  // to be attached. If a ReferenceGrant does not allow this reference, the
+  // "ResolvedRefs" condition MUST be set to False for this listener with the
+  // "RefNotPermitted" reason.
+  //
+  // CertificateRef can reference to standard Kubernetes resources, i.e.
+  // Secret, or implementation-specific custom resources.
+  //
+  // Support: Implementation-specific
+  ClientCertificateRef SecretObjectReference `json:"clientCertificateRef,omitempty"`
+}
+```
+
+#### Limitations
+
+Configuring client certificate on the service level may result in multiple Gateways
+from different vendors sharing the same identity when connecting to a single service.
+
+However, given that:
+
+1. per-service certificate configuration is a niche configuration, required to support
+application-specific compatbility corner-cases
+1. cross-namespace service usage is protected by the reference grants
+
+this limitations should consitute an acceptable trade off.
+
+The possible need for more fine-grained settings can be mitigated by allowing shared 
+policies to selectively target specific Gateway instances, GatewayClasses or gateway 
+namespaces.
+
+### SANs on BackendTLSPolicy
+
+This change enables the certificate to have a different identity than the SNI
+(both are currently tied to the hostname field). This is particularly useful
+when using SPIFFE, which relies on URI Subject Names which are not valid SNIs 
+as per https://www.rfc-editor.org/rfc/rfc6066.html#section-3.
+
+In such case either connection properties or an arbitrary SNI, like cluster-local 
+service name could be used for certificate selection, while the identity validation
```

Comment:
Should hostname still be a required field and used for SNI? Or are you suggesting that it's optional and SNI can be _anything_ when SANs are specified?

Reply:
It cannot by anything, because it needs to be deterministic what we send to the backend. So yes hostname is still required.